stdlib/public/core/UnsafeRawPointer.swift.gyb

```diff
@@ -437,33 +446,32 @@ public struct Unsafe${Mutable}RawPointer : Strideable, Hashable, _Pointer {
   /// `UnsafeMutablePointer.allocate(capacity:)` static method instead.
   ///
   /// - Parameters:
-  ///   - size: The number of bytes to allocate. `size` must not be negative.
-  ///   - alignedTo: The alignment of the new region of allocated memory, in
+  ///   - byteCount: The number of bytes to allocate. `byteCount` must not be negative.
+  ///   - alignment: The alignment of the new region of allocated memory, in
   ///     bytes.
   /// - Returns: A pointer to a newly allocated region of memory. The memory is
   ///   allocated, but not initialized.
   @_inlineable
   public static func allocate(
-    bytes size: Int, alignedTo: Int
+    byteCount: Int, alignment: Int
   ) -> UnsafeMutableRawPointer {
     return UnsafeMutableRawPointer(Builtin.allocRaw(
-        size._builtinWordValue, alignedTo._builtinWordValue))
+        byteCount._builtinWordValue, alignment._builtinWordValue))
   }
 %  end # mutable
 
-  /// Deallocates memory referenced by the pointer with the specified size and
-  /// alignment.
+  @available(*, deprecated, message: "Swift currently only supports freeing entire heap blocks, use deallocate() instead")
+  public func deallocate(bytes _: Int, alignedTo _: Int) { 
+    self.deallocate()
+  }
```

Comment:
In addition to deprecating, do we also need to obsolete this in 5?

Reply:
i don’t know,,, the way i’ve always used `@available` is to deprecate stuff that’s being removed and unavailable–rename stuff that’s been renamed

Reply:
You can supply multiple `@available` attributes - that attribute is the first one [documented here](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/Attributes.html).

Reply:
okay but it doesn’t say what the difference between unavailable and obsoleted and renamed is or *when* to use each one. i’ve just been going off of the example they have midway down the page

    @available(*, unavailable, renamed: "MyRenamedProtocol")

Reply:
Ok, so the swift-book says that you can use `renamed` with `unavailable`, but it actually works with the other availability attributes as well. I *think* we typically want to use `obsoleted: 5, renamed:` so that we have source compatibility in Swift 4 mode but aren't forced to provide those symbols in Swift 5. I'll need to check with @airspeedswift to be sure.

Reply:
does that mean the legacy function implementations have to stay instead of being replaced with `fatalError("unreachable")`??

Reply:
i changed all the `unavailable`s to `obsoleted`s and i did my best to forward them to the actual implementations